crates/apub/src/protocol/objects/page.rs

```diff
@@ -124,6 +125,37 @@ impl Attachment {
       _ => None,
     }
   }
+
+  pub(crate) async fn as_markdown(&self, context: &Data<LemmyContext>) -> LemmyResult<String> {
+    let (url, name, is_image) = match self {
+      Attachment::Image(i) => (i.url.clone(), i.name.clone(), true),
+      Attachment::Document(d) => {
+        let is_image = d
+          .media_type
+          .as_ref()
+          .is_some_and(|media| media.starts_with("video") || media.starts_with("image"));
+
+        (d.url.clone(), d.name.clone(), is_image)
+      }
+      Attachment::Link(l) => {
+        let is_image = l
+          .media_type
+          .as_ref()
+          .is_some_and(|media| media.starts_with("video") || media.starts_with("image"));
```

Comment:
Would be easier to return media_type directly and check for video/image outside the match.